src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/AtmosphericDrag.cpp

```diff
@@ -95,23 +95,31 @@ SatelliteSystem AtmosphericDrag::getSatelliteSystem() const
     return satelliteSystem_;
 }
 
+void AtmosphericDrag::declareCoordinates(const Shared<CoordinatesBroker>& coordinatesBroker)
+{
+    this->positionIndex_ = coordinatesBroker->addSubset(CoordinatesSubset::Position());
+    this->velocityIndex_ = coordinatesBroker->addSubset(CoordinatesSubset::Velocity());
+}
+
 void AtmosphericDrag::applyContribution(
     const Dynamics::StateVector& x, Dynamics::StateVector& dxdt, const Instant& anInstant
 ) const
 {
     (void)anInstant;
 
+    Vector3d positionCoordinates = Vector3d(x[positionIndex_], x[positionIndex_ + 1], x[positionIndex_ + 2]);
```

Comment:
Use size from the `coordinatesSubset` (i.e. 3 here) and slicing to make it more generic? That might be part of the next step, if so feel free to discard